### PR TITLE
fix(backend): automatically handle default card reassignment on card deletion

### LINKED ISSUE #251: fix(backend): automatically handle default card reassignment on card deletion
## Summary
Prevents database inconsistencies and profile breakage by restricting or gracefully handling the deletion of a user's active default profile card.

## Contexts
Currently, the `DELETE /cards/:id` endpoint allows a user to delete any card belonging to them. If they delete their active default card (`isDefault: true`), the user profile is left without a default fallback state. This breaks core external entry points, such as primary QR code resolution and the public-facing web backup route (`apps/web`), which depend on a valid default card.

## Tasks
- [x] Intercept `DELETE /cards/:id` requests to check if the target card has `isDefault: true`.
- [x] Implement fallback logic: if other cards exist, automatically promote the oldest remaining card to `isDefault: true`.
- [x] Prevent deletion or return a descriptive error if it is the user's last remaining card (or auto-create a baseline fallback).
- [x] Add automated unit and integration tests covering the new deletion behavior.

## Acceptance Criteria
- [x] Attempting to delete a default card gracefully reassigns the default flag to an existing sibling card.
- [x] Deleting a non-default card leaves the default card unaffected.
- [x] Back-end unit tests added and pass cleanly (`pnpm test:backend`).

## Area
`backend`

## Difficulty
Medium

### PR DESCRIPTION
**Pull request by mramansayyad**
## Description
This PR addresses **Issue #251**, resolving a critical bug where deleting an active default card would leave a user's profile without a fallback, thereby breaking core external entry points like the primary QR code resolution and the public-facing web backup routes.

## Changes Made
- **Intercept Default Card Deletion:** Modified the `DELETE /cards/:id` endpoint in `apps/backend/src/routes/cards.ts` to intercept the deletion of the active default card.
- **Graceful Fallback Logic:** Implemented an automated reassignment strategy. When the `isDefault: true` card is deleted, the backend seamlessly promotes the oldest remaining card to be the new default.
- **Last Card Protection:** Added logic to prevent users from accidentally deleting their last remaining card, returning a descriptive `400 Bad Request` instead to ensure a baseline fallback is always available.
- **Comprehensive Testing:** Added a dedicated test suite (`apps/backend/src/__tests__/cards.test.ts`) covering all edge cases (deletion of standard cards, reassignment for default cards, and blocking the deletion of the final card).

## Related Issues
Fixes #251

## Testing
- [x] Back-end unit tests added and pass cleanly.
- [x] Tested locally mocking Prisma behaviors to assert standard deletion, fallback logic, and deletion limits.

## Checklist
- [x] Attempting to delete a default card gracefully reassigns the default flag to an existing sibling card.
- [x] Deleting a non-default card leaves the default card unaffected.
- [x] Code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: apps/backend/src/routes/cards.ts
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts:1]: Can we wrap the query inside the try catch for better error handling.

File: apps/backend/src/routes/cards.ts [hunk 1]
```diff
@@ -1,154 +1,158 @@
...
+      }));
+    } catch (error) {
+      return handleDbError(error, request, reply);
+    }
   });
 
   // ─── Create Card ───
 
-  app.post('/', async (request: FastifyRequest, reply: FastifyReply): Promise<object> => {
+  app.post('/', async (request: FastifyRequest, reply: FastifyReply): Promise<Card | void> => {
```
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts]: Can we add request schema here

File: apps/backend/src/routes/cards.ts [hunk 2]
```diff
@@ -38,7 +38,21 @@ export async function cardRoutes(app: FastifyInstance): Promise<void> {
...
+        properties: {
+          title: { type: 'string', minLength: 1, maxLength: 100 },
+          linkIds: {
+            type: 'array',
+            items: { type: 'string', format: 'uuid' }
+          }
+        }
+      }
+    }
+  }, async (request: FastifyRequest, reply: FastifyReply): Promise<Card | void> => {
```
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts]: Fastify typed request schema instead of manually typed.

File: apps/backend/src/routes/cards.ts [hunk 3]
```diff
@@ -38,7 +52,21 @@ export async function cardRoutes(app: FastifyInstance): Promise<void> {
 
   // ─── Create Card ───
 
-  app.post('/', async (request: FastifyRequest, reply: FastifyReply): Promise<Card | void> => {
+  app.post('/', {
```
Comment (Harxhit) [inline apps/backend/src/routes/cards.ts]: Remove manually typed request schema please.

**Review by Harxhit — APPROVED** [review]
